fuel-merkle/src/sparse/merkle_tree.rs

```diff
@@ -33,6 +33,57 @@ impl<StorageError> From<StorageError> for MerkleTreeError<StorageError> {
     }
 }
 
+/// The safe Merkle tree storage key prevents Merkle tree structure manipulations.
+/// The type contains only one constructor that hashes the storage key.
+#[derive(Debug, Clone, Copy)]
+pub struct MerkleTreeKey(Bytes32);
+
+impl MerkleTreeKey {
+    /// The safe way to create a `Self`. It hashes the `storage_key`, making
+    /// it entirely random and preventing SMT structure manipulation.
+    pub fn new<B>(storage_key: B) -> Self
+    where
+        B: AsRef<[u8]>,
+    {
+        use digest::Digest;
+        let mut hash = sha2::Sha256::new();
+        hash.update(storage_key.as_ref());
+        let hash = hash
+            .finalize()
+            .try_into()
```

Comment:
Do we really need a try_into here? I thought finalize returns a `Bytes32` already

Reply:
`finalize` actually returns a `GenericArray` which we have to transform into a native array. We end up doing this `finalize().try_into()...` pattern in a few places because of this.

https://docs.rs/sha2/latest/sha2/trait.Digest.html#tymethod.finalize

Reply:
ah ok, I assumed this was using the hasher from fuel-crypto